tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -453,6 +449,60 @@ public:
    void PrintInfo(std::ostream &output) const;
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RFieldDescriptorBuilder
+\ingroup NTuple
+\brief A helper class for piece-wise construction of an RFieldDescriptor
+*/
+// clang-format on
+class RFieldDescriptorBuilder {
+private:
+   RFieldDescriptor fField = RFieldDescriptor();
+public:
+   RFieldDescriptorBuilder() = default;
+   /// Make an RFieldDescriptorBuilder based off of an existing field descriptor.
+   explicit RFieldDescriptorBuilder(const RFieldDescriptor& fieldDesc) : fField(fieldDesc) {}
+   template<typename AsFieldDescriptor>
```

Comment:
Can't we just use RFieldBase` here instead of the template?

Reply:
Yes, perhaps I went a bit overboard here :)